File: functions/Petshop.Shared/Models/Pet.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Petshop.Shared.Models;

[Table("pets")]
public class Pet
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required(ErrorMessage = "Nome do pet é obrigatório")]
    [StringLength(100, MinimumLength = 2, ErrorMessage = "Nome do pet deve ter entre 2 e 100 caracteres")]
    [Column("nome")]
    public string Nome { get; set; } = string.Empty;

    [Required(ErrorMessage = "Tipo de pet é obrigatório")]
    [StringLength(20)]
    [Column("tipo")]
    public string Tipo { get; set; } = string.Empty; // cao, gato, passaro, coelho, outro

    [Required(ErrorMessage = "Raça é obrigatória")]
    [Column("raca")]
    public string Raca { get; set; } = string.Empty;

    [Required(ErrorMessage = "Idade é obrigatória")]
    [Range(0, 30, ErrorMessage = "Idade deve estar entre 0 e 30 anos")]
    [Column("idade")]
    public int Idade { get; set; }

    [Range(0.1, 100.0, ErrorMessage = "Peso deve estar entre 0.1 e 100 kg")]
    [Column("peso")]
    public double? Peso { get; set; }

    [Required(ErrorMessage = "Sexo do pet é obrigatório")]
    [StringLength(1)]
    [Column("sexo")]
    public string Sexo { get; set; } = string.Empty; // M, F

    [Column("castrado")]
    public bool Castrado { get; set; } = false;

    [StringLength(500)]
    [Column("observacoes")]
    public string? Observacoes { get; set; }

    // Necessidades especiais
    [Column("tem_alergia")]
    public bool TemAlergia { get; set; } = false;

    [Column("precisa_medicacao")]
    public bool PrecisaMedicacao { get; set; } = false;

    [Column("comportamento_agressivo")]
    public bool ComportamentoAgressivo { get; set; } = false;

    // Relacionamentos
    [Required]
    [Column("cliente_id")]
    public long ClienteId { get; set; }

    [ForeignKey("ClienteId")]
    public Cliente? Cliente { get; set; }

    public ICollection<Agendamento> Agendamentos { get; set; } = new List<Agendamento>();
}

File: functions-java/petshop-functions-shared/src/main/java/com/petshop/functions/shared/model/Pet.java
package com.petshop.functions.shared.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "pets")
public class Pet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Nome do pet é obrigatório")
    @Size(min = 2, message = "Nome do pet deve ter no mínimo 2 caracteres")
    @Column(nullable = false)
    private String nome;

    @NotBlank(message = "Tipo de pet é obrigatório")
    @Column(nullable = false, length = 20)
    private String tipo; // cao, gato, passaro, coelho, outro

    @NotBlank(message = "Raça é obrigatória")
    @Column(nullable = false)
    private String raca;

    @NotNull(message = "Idade é obrigatória")
    @Min(value = 0, message = "Idade não pode ser negativa")
    @Max(value = 30, message = "Idade não pode ser maior que 30 anos")
    @Column(nullable = false)
    private Integer idade;

    @DecimalMin(value = "0.0", inclusive = false, message = "Peso deve ser maior que zero")
    @DecimalMax(value = "100.0", message = "Peso não pode ser maior que 100kg")
    private Double peso;

    @NotBlank(message = "Sexo do pet é obrigatório")
    @Column(nullable = false, length = 1)
    private String sexo; // M, F

    @Column(nullable = false)
    private Boolean castrado = false;

    @Column(length = 500)
    private String observacoes;

    @Column(name = "tem_alergia")
    private Boolean temAlergia = false;

    @Column(name = "precisa_medicacao")
    private Boolean precisaMedicacao = false;

    @Column(name = "comportamento_agressivo")
    private Boolean comportamentoAgressivo = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @OneToMany(mappedBy = "pet", cascade = CascadeType.ALL)
    private List<Agendamento> agendamentos = new ArrayList<>();

    // Construtores
    public Pet() {
    }

    public Pet(String nome, String tipo, String raca, Integer idade, Double peso, String sexo, 
               Boolean castrado, String observacoes) {
        this.nome = nome;
        this.tipo = tipo;
        this.raca = raca;
        this.idade = idade;
        this.peso = peso;
        this.sexo = sexo;
        this.castrado = castrado;
        this.observacoes = observacoes;
    }

    // Getters e Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getRaca() {
        return raca;
    }

    public void setRaca(String raca) {
        this.raca = raca;
    }

    public Integer getIdade() {
        return idade;
    }

    public void setIdade(Integer idade) {
        this.idade = idade;
    }

    public Double getPeso() {
        return peso;
    }

    public void setPeso(Double peso) {
        this.peso = peso;
    }

    public String getSexo() {
        return sexo;
    }

    public void setSexo(String sexo) {
        this.sexo = sexo;
    }

    public Boolean getCastrado() {
        return castrado;
    }

    public void setCastrado(Boolean castrado) {
        this.castrado = castrado;
    }

    public String getObservacoes() {
        return observacoes;
    }

    public void setObservacoes(String observacoes) {
        this.observacoes = observacoes;
    }

    public Boolean getTemAlergia() {
        return temAlergia;
    }

    public void setTemAlergia(Boolean temAlergia) {
        this.temAlergia = temAlergia;
    }

    public Boolean getPrecisaMedicacao() {
        return precisaMedicacao;
    }

    public void setPrecisaMedicacao(Boolean precisaMedicacao) {
        this.precisaMedicacao = precisaMedicacao;
    }

    public Boolean getComportamentoAgressivo() {
        return comportamentoAgressivo;
    }

    public void setComportamentoAgressivo(Boolean comportamentoAgressivo) {
        this.comportamentoAgressivo = comportamentoAgressivo;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public List<Agendamento> getAgendamentos() {
        return agendamentos;
    }

    public void setAgendamentos(List<Agendamento> agendamentos) {
        this.agendamentos = agendamentos;
    }
}
